### ISSUE TITLE
error plotting when spectrum has gaps (zero value error)

### ISSUE THREAD

**Issue by shimwell** [fusion-energy/spectrum_plotter#22]
overly aggressive weight windows can cause the spectrum to have gaps. While this is not ideal the package should properly handle this without crashing.

- [ ] add test case to mimic gappy spectrum and reproduce error
- [ ] add solution to handle gappy spectrum

**Comment by shimwell** [comment]
```python
File "/home/spectrum_plotter/spectrum_plotter/core.py", line 319, in add_spectra_to_plot
    y=y + y_err,
UnboundLocalError: local variable 'y_err' referenced before assignment
```

**Comment by py1sl** [comment]
not sure this is a gappy spectrum issue, the error was for plotly not having a if len(spectra=3) around the traces for the std dev.
might be gappy spectra + log scale will cause an issue

**Comment by shimwell** [comment]
Yes that sounds familiar, the spectrum had gaps it it which I guess were either 0 or NaN values. Thanks for figuring that out.
